provider/core/src/lib.rs

```diff
@@ -248,7 +248,23 @@ pub mod prelude {
     pub use crate::DynamicDataProvider;
     #[doc(no_inline)]
     pub use crate::KeyedDataMarker;
+    #[doc(no_inline)]
+    #[doc(no_inline)]
+    pub use icu_locid::extensions::unicode::{
+        key as unicode_extension_key, Key as UnicodeExtensionKey,
+    };
+    #[doc(no_inline)]
+    pub use icu_locid::extensions::unicode::{
+        value as unicode_extension_value, Value as UnicodeExtensionValue,
+    };
+    #[doc(no_inline)]
+    pub use icu_locid::{langid, LanguageIdentifier};
 
+    #[cfg(feature = "experimental")]
+    pub use icu_locid::extensions::private::{subtag as aux_key, Subtag as AuxiliaryKey};
+
+    #[doc(hidden)]
+    pub use icu_locid;
```

Comment:
Suggestion: keep this as the only re-export, without re-exporting the individual types.

- We don't re-export things from `yoke` and `zerofrom`
- Risk of name clashes
- Avoid tying the exact set of re-exports to this crate's semver

Reply:
The types I'm reexporting here are fundamentally needed to construct `DataLocale`s. Just look how many imports this removes all over the place.

Reply:
I wish you had just done a clean re-export of `icu_locid` because I would approve that in a heartbeat. But since you're putting novel names in the prelude, and the names could potentially clash with other names, I hesitate, even if I like the outcome.

Reply:
What would `LanguageIdentifier`, `UnicodeExtensionKey`, and `UnicodeExtensionValue` possibly clash with?